rust/backend/daemon/src/features.rs

```diff
@@ -1,16 +1,230 @@
+            /*let link_id = jni_add_local.attach(
+                Some("AddLocalRef"),
+                offset,
+                target,
+                None
+            )?;
+            self.trace_add_local = Some(jni_add_local.take_link(link_id)?);*/
+
+
+        }
+
+        if self.trace_del_local.is_none() {
+            let jni_del_local: &mut UProbe = ebpf
+                .program_mut("trace_del_local")
+                .ok_or(EbpfError::ProgramError(
+                    aya::programs::ProgramError::InvalidName {
+                        name: "trace_del_local".to_string(),
+                    },
+                ))?
+                .try_into()?;
+
+            let offset = todo!("get offset of symbol");
+            let target = todo!("get absolute path to library/ binary");
+
+            /*let link_id = jni_add_global.attach(
+                Some("AddGlobalRef"),
+                offset,
+                target,
+                None
+            )?;
+            self.trace_add_global = Some(jni_add_global.take_link(link_id)?);*/
+        }
+
+        if self.trace_add_global.is_none() {
+            let jni_add_global: &mut UProbe = ebpf
+                .program_mut("trace_add_global")
+                .ok_or(EbpfError::ProgramError(
+                    aya::programs::ProgramError::InvalidName {
+                        name: "trace_add_global".to_string(),
+                    },
+                ))?
+                .try_into()?;
+
+            let offset = todo!("get offset of symbol");
+            let target = todo!("get absolute path to library/ binary");
+
+            /*let link_id = jni_add_global.attach(
+                Some("AddGlobalRef"),
+                offset,
+                target,
+                None
+            )?;
+            self.trace_add_global = Some(jni_add_global.take_link(link_id)?);*/
+        }
+
+        if self.trace_del_global.is_none() {
+            let jni_del_global: &mut UProbe = ebpf
+                .program_mut("trace_del_global")
+                .ok_or(EbpfError::ProgramError(
+                    aya::programs::ProgramError::InvalidName {
+                        name: "trace_del_global".to_string(),
+                    },
+                ))?
+                .try_into()?;
+
+            let offset = todo!("get offset of symbol");
+            let target = todo!("get absolute path to library/ binary");
+
+            /*let link_id = jni_del_global.attach(
+                Some("DeleteGlobalRef"),
+                offset,
+                target,
+                None
+            )?;
+            self.trace_del_global = Some(jni_del_global.take_link(link_id)?);*/
+        }
+
+        Ok(())
+    }
+
+    pub fn detach(&mut self, ebpf: &mut Ebpf) -> Result<(), EbpfError> {
+        let _ = self.trace_add_local.take();
+        let _ = self.trace_del_local.take();
+        let _ = self.trace_add_global.take();
+        let _ = self.trace_del_global.take();
+
```

Comment:
this wont work if you are using LinkIds, the link ids have to be detached manually. If you have Links then they are detached on drop